File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        ordered_map = dict()
        for idx, val in enumerate(order):
            ordered_map[val] = idx
            
        for i in range(len(words) - 1):
            for j in range(len(words[i])):
                if j >= len(words[i + 1]): return False
            
                if words[i][j] != words[i+1][j]:
                    if ordered_map[words[i][j]] > ordered_map[words[i+1][j]]: return False
                    break
        return True

File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& words, string order) {
        unordered_map<char, int> lookup;
        for (int i = 0; i < order.size(); ++i) lookup[order[i]] = i;
        
        for (int i = 0; i < words.size()-1; ++i){
            for (int j = 0; j < words[i].size(); ++j){
                if (j >= words[i+1].size()) return false;
                
                if (words[i][j] != words[i+1][j]){
                    if (lookup[words[i][j]] > lookup[words[i+1][j]]) return false;    
                    break;
                }
            }
        }
        
        return true;
    }
};